src/components/Card.js

```diff
@@ -1,16 +1,38 @@
+import Popup from "./Popup.js";
 
 export default class Card {
-    constructor(data, cardSelector, handleImageClick) {
+    constructor(data, cardSelector, handleImageClick, handleDeleteButton, handleCardLike, handleCardDislike) {
         this._name = data.name;
         this._link = data.link;
         this._cardSelector = cardSelector;
         this._handleImageClick = handleImageClick;
+        this._handleDeleteButton = handleDeleteButton;
+        this._handleCardLike = handleCardLike;
+        this._handelCardDislike = handleCardDislike;
+        this._deleteSubmitHandler = this._handleDeleteSubmit.bind(this);// Bind here
+        this._imageSubmitHandler = this._handleImageSubmit.bind(this);
+        this._id = data._id;
+        this._isLiked = data.isLiked;
     }
 
     _setEventListeners() {
-        this._likeButton.addEventListener('click', () => this._handleLikeClick());
+        this._likeButton.addEventListener('click', () => {
+            if(this._isLiked){
+                this._handelCardDislike(this);
+            } else {
+                this._handleCardLike(this);
+            }
+        });
+
+        this._deleteButton.addEventListener('click', () => {
+            console.log('trash clicked');
+            this._modalDelete.classList.add('modal_opened');
+            const messageElement = document.querySelector('#delete-button');
+            messageElement.textContent = 'Yes';
+            document.addEventListener('keydown', this.escPress.bind(this));
+            this._deleteSubmit.addEventListener('submit', this._deleteSubmitHandler);
```

Comment:
Please, delete it

Every modal should be created using `Popup`. And submit handler should be added inside the class